### DISCUSSION TITLE
verify-segment find many lost file ,then how to recovery

### DISCUSSION THREAD

**Discussion by wwli05** [milvus-io/milvus#30842] [Q&A and General discussion]
-Milvus version : 2.3.0
our milvus cluster has dozens of collection, already run some weeks, the problem is : the segment flush every 10 minutes, but in one flush , all the columns flushed file is lost from minio.
the folder structure in minio is like this:
```
 segementid/0/
                       21:00  fileid
                       21:10   fileid   --21:20's file is lost
                       21:30   fileid
                    /1/
                       21:00  fileid
                       21:10   fileid   --21:20's file is lost
                       21:30   fileid                
```   
we use verify-segment find file not exists in minio files/insert_log/xx/xxx, because of these lost files, the indexnode continueous restart as of "failed to build index"  getobjectsize 404.  
meanwhile birdwatcher backup hangs with info ... metrics
so my question is:  how to discard those lost files let the build index success?

**Comment by yhmo** [comment]
So far as we know, two milvus servers running on the same S3 data path, can clean data files of each other and lead to data lost.

**Comment by wwli05** [comment]
we use minio,  different cluster use its own minio.
is there any other command to repair the meta after verify-segment found corrupt info?

**Reply by xiaofan-luan** [reply]
So far you need to remove the files from etcd with birdwatcher to recover

**Reply by yhmo** [reply]
Are the two clusters using the same etcd service?

**Reply by wwli05** [reply]
no, the cluster has its own etcd, and they have worked well  for some days with no change/no new deploy

**Reply by xiaofan-luan** [reply]
remove binlog to remove the unexist logs.

**Comment by wwli05** [comment]
at 2.3.0,https://github.com/milvus-io/milvus/blob/v2.3.0/internal/datanode/flush_manager.go#L520
```
	m.handleInsertTask(segmentID, &flushBufferInsertTask{
		ChunkManager: m.ChunkManager,
		data:         kvs,
	}, field2Insert, field2Stats, flushed, dropped, pos)

	metrics.DataNodeEncodeBufferLatency.WithLabelValues(fmt.Sprint(paramtable.GetNodeID())).Observe(float64(tr.ElapseSpan().Milliseconds()))
	return stats, nil
```
after handleInsertTask, no where check whether the insert task has error or not, so i think if error happended, but it will continue to write meta info, so inconsistent like i met will happen.  could you confirm this?

**Comment by xiaofan-luan** [comment]
> at 2.3.0,https://github.com/milvus-io/milvus/blob/v2.3.0/internal/datanode/flush_manager.go#L520
> 
> ```
> 	m.handleInsertTask(segmentID, &flushBufferInsertTask{
> 		ChunkManager: m.ChunkManager,
> 		data:         kvs,
> 	}, field2Insert, field2Stats, flushed, dropped, pos)
> 
> 	metrics.DataNodeEncodeBufferLatency.WithLabelValues(fmt.Sprint(paramtable.GetNodeID())).Observe(float64(tr.ElapseSpan().Milliseconds()))
> 	return stats, nil
> ```
> 
> after handleInsertTask, no where check whether the insert task has error or not, so i think if error happended, but it will continue to write meta info, so inconsistent like i met will happen. could you confirm this?

If flush failed, datanode will panic.
this happened inside handleInsert task.
On master branch we already refined that part. And on early version of 2.3 this happens and we already fixed some of these issues in later 2.3.x
